### PR TITLE
Implemented getOutputElementType

### PR DESCRIPTION
**Pull request by Pey-crypto**
Implemented Method on c++ side.
Updated typescript definitions.
Created unit tests.
For Issue [https://github.com/openvinotoolkit/openvino/issues/25406](https://github.com/openvinotoolkit/openvino/issues/25406)

Resolved merge errors

### CODE REVIEW

File: thirdparty/open_model_zoo
Comment (rkazants) [inline thirdparty/open_model_zoo:1]: please revert this change, it is not needed

File: package-lock.json
Comment (rkazants) [inline package-lock.json:1]: not sure that any packaging file is needed for this feature. please check

**Comment by Pey-crypto** [comment]
can i add `if (info.Length() == 1) { error } ` into the getOutputElementType method or is there an alternative way to check? Adding this would solve the errors reported by the second subtests. @almilosz

File: src/bindings/js/node/src/model_wrap.cpp [hunk 1]
```diff
@@ -2,11 +2,11 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "node/include/model_wrap.hpp"
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:4]:
```suggestion
#include "node/include/model_wrap.hpp"

```

**Review by almilosz — CHANGES_REQUESTED** [review]
Hi Pey-crypto!
Answering your question, you cannot add `if (info.Length() == 1) { error } ` check. This is done by `ov::js::validate()` for each correct signature so adding this check would be redundant. 
The second subtest returns errors because you provided incorrect error messages. I suggest making it shorter (see code suggestion). 

You still need to apply my suggestion from the last review I made about method description and remove extra changes from:
 - __.gitignore_
 - _package.json_
 - _thirdparty/open_model_zoo__

If you need more help or detailed guidance you can find me on Discord: `_almilosz`

File: package.json
Comment (ilya-lavrenov) [inline package.json:1]: do we need it?

Comment (almilosz) [reply package.json:1]: We don't

Comment (Pey-crypto) [reply package.json:1]: It was a mistake from my end, i was trying to setup the unit tests and executed npm cmd in the main directory. The file has been removed in the new commit.

Comment (almilosz) [reply package.json:1]: I think you didn't push the new commit that removes _package.json_

You also need to revert changes from __.gitignore_ and _thirdparty/open_model_zoo__
If you need any help with that, you can find me on Discord _almilosz

**Review by p-wysocki — COMMENTED** [review]
Hello @Pey-crypto! It seems that there are 2 issues left:

- Changes to `.gitignore` file
Let's keep the PR limited to the JS API component. When you go to "Files Changes" in the PR view you can see that something has been changed in `.gitignore`. Let's revert that change and align the file with OpenVINO master. You can do it with `git checkout upstream/master -- .gitignore`, assuming the main OpenVINO repository remote is called `upstream`, which you can check using `git remote -v`.

- Changes to `open_model_zoo` submodule
`open_model_zoo` is a git submodule, which means that it's a GitHub repository "embedded" into OpenVINO repository as a dependency. For some reason your PR shows that changes have been made to it, most likely due to a gone wrong merge. In order to undo that, you need to:

1. go to `openvino_repo/thirdparty/open_model_zoo`
2. run `git fetch --all` to get all the data on exisintg `open_model_zoo` branches and commits
3. check which `open_model_zoo` commit sha is currently on OpenVINO master by going to `https://github.com/openvinotoolkit/openvino/tree/master/thirdparty`, clicking on `open_model_zoo` and copying the commit hash from URL
4. run `git checkout <commit_sha_you_just_copied>`


Once you do both these things you can just commit them and push. Once it's done the PR will be fixed. :)

File: src/bindings/js/node/include/model_wrap.hpp [hunk 2]
```diff
@@ -108,6 +108,19 @@ class ModelWrap : public Napi::ObjectWrap<ModelWrap> {
...
+     * @return Napi::String representing the element type of the requested output.
+     *
+     */
+    Napi::Value get_output_element_type(const Napi::CallbackInfo& info);
+
+    /**
+     * @brief Helper function to access model output elements types.
+     * @return Napi::String representing the element type of the requested output.
+     *
+     */
```
Comment (p-wysocki) [inline src/bindings/js/node/include/model_wrap.hpp:123]: The method description should be placed only once, above the declaration. This one can be deleted.
```suggestion
```

**Review by almilosz — APPROVED** [review]

File: src/bindings/js/node/lib/addon.ts [hunk 3]
```diff
@@ -234,9 +234,14 @@ interface Model {
...
   getOutputSize(): number;
+  /**
+   * It gets the element type of a specific output of the model.
+   * @param index The index of the output.
+   */
+  getOutputElementType(index: number): string;
   /**
    * It gets the input of the model.
-   * If a model has more than one input, this method throws an exception.
+   * if a model has more than one input, this method throws an exception.
```
Comment (almilosz) [inline src/bindings/js/node/lib/addon.ts:244]:
```suggestion
   * If a model has more than one input, this method throws an exception.
```

**Comment by almilosz** [comment]
build_jenkins

**Pull request merged**